mne/io/meas_info.py

```diff
@@ -589,12 +589,33 @@ def _check_consistency(self):
             raise RuntimeError('bad channel(s) %s marked do not exist in info'
                                % (missing,))
         meas_date = self.get('meas_date')
-        if meas_date is not None and (
-                not isinstance(self['meas_date'], tuple) or
-                len(self['meas_date']) != 2):
-            raise RuntimeError('info["meas_date"] must be a tuple of length '
-                               '2 or None, got "%r"'
-                               % (repr(self['meas_date']),))
+        if meas_date is not None:
+            if (not isinstance(self['meas_date'], tuple) or
+                    len(self['meas_date']) != 2):
+                raise RuntimeError('info["meas_date"] must be a tuple '
+                                   'of length 2 or None, got "%r"'
+                                   % (repr(self['meas_date']),))
+            if (meas_date[0] < np.iinfo('>i4').min or
+                    meas_date[0] > np.iinfo('>i4').max):
+                raise RuntimeError('info["meas_date"] must be between "%r" '
+                                   'and "%r", got "%r"'
+                                   % (repr((np.iinfo('>i4').min, 0)),
+                                      repr((np.iinfo('>i4').max, 0)),
+                                      repr(self['meas_date']),))
+
+        for key in ('file_id', 'meas_id'):
+            value = self.get(key)
+            if value is not None:
```

Comment:
```suggestion
            if key in self:
```

Reply:
avoids the non-kosher way of checking if key exists in dict

Reply:
I would also do:

```
if key in self:
    value = self[key]
```

Reply:
the problem is that `key in self` is True but `value is None`

Reply:
ok got it !